app/controllers/isa_assays_controller.rb

```diff
@@ -14,7 +14,8 @@ def create
     update_sharing_policies @isa_assay.assay
     @isa_assay.assay.contributor = current_person
     @isa_assay.sample_type.contributor = User.current_user.person
-
+    @isa_assay.assay.creators = get_creator_ids_for_isa_assays(params).collect{ |creator_id| Person.find(creator_id) }
+    @isa_assay.assay.other_creators = params[:assay][:other_creators] if params[:assay].respond_to? :other_creators
```

Comment:
you shouldn't be needing to jump through these hoops if the param attributes are correct, which they should be if using the `assets/manage_specific_attributes` partial, which you seem to be. It should be being set when calling `IsaAssay.new(isa_assay_params)` or maybe calling update on `@isa_assay.assay`

I would double check they are getting passed through and approved in `isa_assay_params`

Reply:
For a reason I cant't figure out the assay_params, are not fully passed to the isa_assay_params.

```ruby
@isa_assay.assay.attributes = isa_assay_params[:assay]
```
Sets the "other_creators" attribute but not the creators.

Reply:
I've  found something peculiar: In the params there is a `params[:study]` and a `params[:isa_study][:study]` which contain different attributes.

```ruby
>> params[:study]
=> #<ActionController::Parameters {"investigation_id"=>"2", "assets_creators_attributes"=>{"1"=>{"pos"=>"1", "creator_id"=>"1", "family_name"=>"De Pelseneer", "given_name"=>"Kevin", "affiliation"=>"Default Institution, Fun University", "orcid"=>""}, "2"=>{"pos"=>"2", "creator_id"=>"3", "family_name"=>"Sheraton", "given_name"=>"Sheryl", "affiliation"=>"", "orcid"=>""}}, "other_creators"=>"Lord Bob Bobbington the third"} permitted: false>
>> params[:isa_study][:study]
=> #<ActionController::Parameters {"title"=>"Test study from Single Page", "investigation_id"=>"2", "description"=>"", "position"=>"", "publication_ids"=>[""], "sop_id"=>"1"} permitted: false>
```
I think that might be the reason the creators aren't linked "out of the box" and I have to manually assign them.

Reply:
I think I know why, in the `_author_form `it uses `form.object`  and bases the field names from that type, so will collapse it down to just `:study`

https://github.com/seek4science/seek/blob/cfb9dd0719bd86084d99219dd3b713109b302d8a/app/views/assets/_author_form.html.erb#L7

Reply:
I think this would be solvable by basing the `field_name `from `form.object_name` as instead, which will the parent, i.e

```
0> form.object_name
=> "isa_assay[assay]"
```

Reply:
Thanks for the fix!
It all works out of the box now.